extensions/keycloak-admin-resteasy-client/runtime/src/main/java/io/quarkus/keycloak/adminclient/ResteasyKeycloakAdminClientRecorder.java

```diff
@@ -126,6 +128,24 @@ public void avoidRuntimeInitIssueInClientBuilderWrapper() {
     // makes media type more specific which ensures that it will be used first
     @Produces(MediaType.APPLICATION_JSON)
     @Consumes(MediaType.APPLICATION_JSON)
-    static class AppJsonQuarkusJacksonSerializer extends QuarkusJacksonSerializer {
+    static final class AppJsonQuarkusJacksonSerializer extends QuarkusJacksonSerializer {
+
+        private final ObjectMapper objectMapper;
+
+        private AppJsonQuarkusJacksonSerializer() {
+            this.objectMapper = new ObjectMapper();
+            // Same like JSONSerialization class. Makes it possible to use admin-client against older
+            // versions of Keycloak server where the properties on representations might be different
+            this.objectMapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+            // The client must work with the newer versions of Keycloak server, which might contain the JSON fields
+            // not yet known by the client. So unknown fields will be ignored.
+            this.objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
```

Comment:
I'm surprised to this both here and in [ResteasyReactiveClientProvider.java](https://github.com/quarkusio/quarkus/pull/43260/files#diff-ea05db94e081f24f3968785ddeb2e62d4246e3916e76a35fbc6446042128030a). `ObjectMapper` is a really heavy object and reusing them should be done if possible.

Reply:
> These 2 extensions are not going to be used together, so they can't share same ObjectMapper instance

Okay, that settles it, thanks!

Reply:
If we are not reusing an `ObjectMapper` we already have, shouldn't we let the Keycloak libraries instantiate and configure the `ObjectMapper` as they see fit?

Or do we absolutely need to pass one?

Reply:
Experience is that Keycloaks provider wasn't native compatible in past. What I did now was necessary in past (you can lookup issues, JSONB was one of them), but I don't know if it is still true, I think we have limited native test coverage for RESTEasy one. Does it matter?

Reply:
Well, it somehow matters given we had to tweak things to have a compatible client. And we're not entirely sure we enabled all the right options. It seems to work for now but what if they change the serialization option in the future?

Now maybe I'm being too cautious but that's something we could miss in the future. And if for instance, we don't serialize something correctly, it might end up having security consequences.

Reply:
Note that if you're on PTO, this can wait early Janiuary :). I'm not on PTO so that's why you see me popping :).

Reply:
> Well, it somehow matters given we had to tweak things to have a compatible client. And we're not entirely sure we enabled all the right options. It seems to work for now but what if they change the serialization options in the future?

note that this code came from @pskopek in the REST client version and I repeatedly asked him if it is fine to do that (not that I got satisfactory answer, this was only _one_ - https://github.com/quarkusio/quarkus/pull/43260#discussion_r1842532591).

>  It seems to work for now but what if they change the serialization options in the future?
> Now maybe I'm being too cautious but that's something we could miss in the future. And if for instance, we don't serialize something correctly, it might end up having security consequences.

1. Absolutely, but we can't know and I asked repeatedly if it is fine. **I agree with you.** We have Keycloak folk integrating Keycloak admin client here, it can't get any better. Only thing we could do is to switch to the KC `JacksonProvider` in the RESTEasy one, but I am pretty sure we don't have test coverage for all the past native issues because there wasn't a good place to put them. I believe that `Keycloak Admin REST Client` is both tested in native and preferred.
2. Even if we do that, we still have manually managed `ObjectMapper` in the REST admin client, so if we need to get it right there, why can't we use same code here?

Reply:
> Now maybe I'm being too cautious but that's something we could miss in the future. And if for instance, we don't serialize something correctly, it might end up having security consequences.

I agree, but the situation in the reactive client is same, so I don't believe it would make a difference here. What would make a difference is Keycloak providing ObjectMapper factory we could re-use in both extensions. @pskopek WDYT?

The idea that Keycloak client is compatible with both previous and future versions of Keycloak makes me nervous as I can't forsee if any incompatibility could result in a vulnerability, but I have already mentioned it before.